rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -140,26 +131,33 @@ public MavenPomDownloader(Map<Path, Pom> projectPoms, HttpSender httpSender, Exe
         this.projectPomsByGav = projectPomsByGav(projectPoms);
         this.httpSender = httpSender;
         this.ctx = MavenExecutionContextView.view(ctx);
-        this.sendRequest = Retry.decorateCheckedFunction(
-                mavenDownloaderRetry,
-                request -> {
-                    int responseCode;
-                    long start = System.nanoTime();
-                    try (HttpSender.Response response = httpSender.send(request)) {
-                        if (response.isSuccessful()) {
-                            return response.getBodyAsBytes();
-                        }
-                        responseCode = response.getCode();
-                    } catch (Throwable t) {
-                        throw new HttpSenderResponseException(t, null);
-                    } finally {
-                        this.ctx.recordResolutionTime(Duration.ofNanos(System.nanoTime() - start));
-                    }
-                    throw new HttpSenderResponseException(null, responseCode);
-                });
         this.mavenCache = this.ctx.getPomCache();
     }
 
+    byte[] sendRequest(HttpSender.Request request) throws HttpSenderResponseException {
+        long start = System.nanoTime();
+        try {
+            return Failsafe.with(retryPolicy).get(() -> {
+                try (HttpSender.Response response = httpSender.send(request)) {
+                    if (!response.isSuccessful()) {
+                        throw new HttpSenderResponseException(null, response.getCode());
+                    }
+                    return response.getBodyAsBytes();
+                }
+            });
+        } catch (Throwable t) {
+            if (t.getCause() != null) {
+                if (t.getCause() instanceof HttpSenderResponseException) {
+                    throw (HttpSenderResponseException) t.getCause();
+                }
+                throw new HttpSenderResponseException(t.getCause(), null);
```

Comment:
I worry a bit about this wrapping any `Throwable.cause` in a `HttpServerResponseException`, since we use the response exception to decide that a server existed on these lines for instance: https://github.com/openrewrite/rewrite/blob/e17faa5222a99b29bbae6c9c20e813faa2d99051/rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java#L753-L754

I think we'd best reserve `HttpServerResponseException` for cases where there is a http response code to pass in, as there's a lot of nested try/catches below that are already hard to reason about, which would be even more difficult if there's `HttpServerResponseException` without http responses.